src/librbd/librbd.cc

```diff
@@ -5087,7 +5087,7 @@ extern "C" int rbd_resize(rbd_image_t image, uint64_t size)
   librbd::ImageCtx *ictx = (librbd::ImageCtx *)image;
   tracepoint(librbd, resize_enter, ictx, ictx->name.c_str(), ictx->snap_name.c_str(), ictx->read_only, size);
   librbd::NoOpProgressContext prog_ctx;
-  int r = ictx->operations->resize(size, true, prog_ctx);
+  int r = ictx->operations->resize(size, false, prog_ctx);
```

Comment:
This would be a backwards incompatible change, breaking e.g. QEMU which still uses `rbd_resize()`.  It's a legacy API -- `rbd_resize2()` takes `allow_shrink` argument and behaves appropriately.

On the CLI, `rbd resize` command has `--allow-shrink` option and refuses to shrink the image if it's not passed.  It doesn't seem like there is anything else to do here.